src/variable.cpp

```diff
@@ -215,7 +215,16 @@ VariablePtr Variable::create(const std::string &name) noexcept
 
 bool Variable::addEquivalence(const VariablePtr &variable1, const VariablePtr &variable2)
 {
-    return (variable1 != nullptr && variable2 != nullptr && variable1->mPimpl->setEquivalentTo(variable2) && variable2->mPimpl->setEquivalentTo(variable1));
+    if (variable1 != nullptr && variable2 != nullptr) {
+        bool canAdd1 = variable1->mPimpl->setEquivalentTo(variable2);
+        bool canAdd2 = variable2->mPimpl->setEquivalentTo(variable1);
+        if (canAdd1 && !canAdd2) {
+            // Remove connection from variable1, since it can't be added to variable2:
+            variable1->mPimpl->unsetEquivalentTo(variable2);
+        }
+        return (canAdd1 && canAdd2);
```

Comment:
```
return canAdd1 && canAdd2;
```

Reply:
I've never understood that one ... just like I've never understood using `++i` instead of `i++` ... sigh.  Done in 2b2884ad.

Reply:
`return` is not a function, so no need for the parentheses.

`++i` vs. `i++`, it's just that the former results in slightly faster code. Truth be told, we will most likely never notice the difference. So, personally I go for `++i` out of habit and because it can only improve the speed of the code.